piksi_multi_cpp/src/device/device.cc

```diff
@@ -31,4 +31,31 @@ int32_t Device::write_redirect(uint8_t* buff, uint32_t n, void* context) {
   return instance->write(std::vector<uint8_t>(&buff[0], &buff[n]));
 }
 
+bool Device::open() {
+  mtx_.lock();
```

Comment:
there's also https://en.cppreference.com/w/cpp/thread/lock_guard

although in this case I favor the clear readability of locking and unlocking, up to you :)